server/actions/InventoryItems.ts

```diff
@@ -0,0 +1,66 @@
+import InventoryItemSchema from 'server/models/InventoryItem'
+import * as MongoDriver from 'server/actions/MongoDriver'
+import { InventoryItem } from 'utils/types'
+import { ApiError } from 'utils/types'
+
+/**
+ * Checks to see if an item is in the inventory and will add to the quantity or
+ * create the item if needed.
+ * @param item
+ * @param quantity of item to add
+ * @returns checksIn Items to the inventory
+ */
+export async function checkInInventoryItem(
+  item: InventoryItem,
+  quantityItem: number
+) {
+  const itemMatches = await MongoDriver.findEntities(InventoryItemSchema, item)
+  if (itemMatches.length) {
+    itemMatches[0].quantity += quantityItem
+    itemMatches[0].attributes?.sort()
+    MongoDriver.updateEntity(
+      InventoryItemSchema,
+      itemMatches[0].id,
+      itemMatches[0]
+    )
+  } else {
+    const createdItem = await MongoDriver.createEntity(
+      InventoryItemSchema,
+      item
+    )
+    createdItem.attributes?.sort()
```

Comment:
Just to not leave anything to chance, this'll need to specifically sort by _id. To do this, you can feed `sort()` an arrow function to compare ids. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). [Here's](https://flaviocopes.com/how-to-sort-array-of-objects-by-property-javascript/) a good example of that in practice

Reply:
Do you want to sort by id and attribute?

Reply:
Sort the attributes by id

Reply:
Let me know if you have questions on this

Reply:
move this before the item is created